### DISCUSSION TITLE
fields array - how to remove items correctly

### DISCUSSION THREAD

**Discussion by n-margalit** [react-hook-form/react-hook-form#7551] [Q&A] [answered]
I'm using rhf and it is amazing, thank you!
using field array, when I remove and add - the new item get the value of the original items in that index (before remove)
[in this code sandbox](https://codesandbox.io/s/rhf-field-array-validation-6llkd) I tried the following scenario:

1. added 3 items children
2. deleted the second item
3. added a new one
4. before editing the new item- got filled with the 3# item from step 1

what does I do incorrectly?

TIA for any help!

**Comment by Moshyfawn** [comment, marked as answer]
I believe, [my answer](https://github.com/react-hook-form/react-hook-form/discussions/7550#discussioncomment-1979877) from #7550 discussion has fixed this issue. Refer to [this comment](https://github.com/react-hook-form/react-hook-form/discussions/7550#discussioncomment-1979877) for explanation _(+ [CSB](https://codesandbox.io/s/rhf-field-array-validation-forked-7u7eh?file=/src/TextField.tsx))_
TL;DR RHF Controller couldn't get the proper values with brackets notation `children[${index}].name`. Should use dot notation instead `children.${index}.name`

**Reply by n-margalit** [reply]
yes it does answer. thanks!
